### PR TITLE
Add functioanlity to preserve data

### PR DESCRIPTION
**Pull request by BHObserver**
In this branch

- **Added functionality to preserve data in local storage**
- **Used JS features**
- **Created a JS object to save data**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -149,13 +149,18 @@ <h2 class="primary color1">I'm always interested in hearing about new projects,
...
+                    <input type="text" name="last_name" required maxlength="30" placeholder="Last Name">
                     <input type="text" name="full_name" required maxlength="30" placeholder="Full Name">
                 </div>
                 <input type="email" name="email" required maxlength="30" placeholder="Email Address">
                 <textarea type="text" name="message" required class="textarea" maxlength="500" placeholder="Hey I'd like to connect too"></textarea>
-                <button type="submit" class="get-in-touch-btn" name="get-in-touch" id="">Get in touch</button>
+                <div class="validation-message-container">
+                    <p class="validation-message"></p>
+                </div>
+                <button type="submit" class="get-in-touch-btn" name="submit">Get in touch</button>
```
Comment (DeliceLydia) [inline index.html]: - [ ] When I reload the page  information should be persistent

Comment (DeliceLydia) [reply index.html]: - [ ] After filling out the form, and not submitting it, the entered values should be saved in the browser but currently its not saved

File: main.js [hunk 2]
```diff
@@ -196,6 +196,77 @@ document.addEventListener('keydown', (event) => {
...
+});
+
+emailInput.addEventListener('input', () => {
+  formData.email = emailInput.value;
+  localStorage.setItem('formDataStorage', JSON.stringify(formData));
+});
+
+messageInput.addEventListener('input', () => {
+  formData.message = messageInput.value;
+  localStorage.setItem('formDataStorage', JSON.stringify(formData));
```
Comment (DeliceLydia) [inline main.js:267]: - [ ] Make sure When the user loads the page, the code checks if there are any saved data in the local storage, and if so the input fields are pre-filled with the data

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Followed git workflow
- No linter errors


## Required Changes ♻️
_Check the comments under the review._



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: index.html (hunk 1 above)
Comment (BHObserver) [reply index.html]: @DeliceLydia I am sorry but I don't understand, it is working properly in my machine. Could you please explain a bit more.

**Review by divinecharlotte — APPROVED** [review]
# Hello @BHObserver 😃 🥳 🎉 🎈 

Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say  👍🏻 ✅ 
### To Highlight 💯


- Good implementation of the required changes 👍 
- Linters are passing ✔ 👍
 




## Congratulations! 🎉 



Cheers, and Happy coding!👏👏👏


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**